### PR TITLE
FEATURE: Adding OpenShift support.

### PR DESCRIPTION
**Pull request by kiuby88**
Some entities and interfaces are adding which are focus on OpenShift support.
Currently, just Java technology is supported in this provided.
Custom env are supported in this version.

Any OS service owner (add-on) is include for now.

TODO:
* Unifying the code of CF and OS support (for avoiding duplicate code).
* Adding PHP and other technologies support.

### CODE REVIEW

**Comment by kiuby88** [comment]
@mbarrientos, could you take a look?

File: deployer/src/main/java/brooklyn/entity/openshift/OSPaasEntityOpenShiftDriver.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+package brooklyn.entity.openshift;
+
+import brooklyn.location.openshift.OpenShiftPaasLocation;
+import com.openshift.client.IOpenShiftConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public abstract class OSPaasEntityOpenShiftDriver implements OSPaasEntityDriver {
```
Comment (mbarrientos) [inline deployer/src/main/java/brooklyn/entity/openshift/OSPaasEntityOpenShiftDriver.java]: Following the same convention for CloudFoundry entities, I think this should be named "PaasEntityOpenShiftDriver", without the "OS" prefix.

Comment (kiuby88) [inline deployer/src/main/java/brooklyn/entity/openshift/OSPaasEntityOpenShiftDriver.java]: could it be better adding the CF to CF-classes?

**Comment by kiuby88** [comment]
@adriannieto I have rebase this branch. Take a look please

**Comment by adriannieto** [comment]
It looks nice, it's ready to merge!

**Comment by kiuby88** [comment]
Thanks a lot @adriannieto

**Comment by perezp** [comment]
Merging... thanks @adriannieto @kiuby88

**Pull request merged**